### PR TITLE
Rain stuff

### PR DESCRIPTION
**Pull request by supsm**
lol

### CODE REVIEW

**Comment by CasuallyMatt** [comment]
funy c++ man bac at it again!

**Comment by supsm** [comment]
:tinypotato:

**Comment by supsm** [comment]
seems like i broke ao lol

**Comment by spiralhalo** [comment]
cloud ceiling is cursed

I'm not merging this lol

File: assets/lumi/shaders/common/shadow.glsl [hunk 1]
```diff
@@ -70,7 +70,7 @@ float calcShadowFactor(in sampler2DArrayShadow shadowMap, vec4 shadowViewPos) {
         float shadowFactor = sampleShadowPCF(shadowMap, shadowCoords.xyz, float(cascade));
     #endif
     
-    return shadowFactor;
+    return min(1, shadowFactor + frx_rainGradient() * 0.8 + frx_thunderGradient() * 0.2);
```
Comment (spiralhalo) [inline assets/lumi/shaders/common/shadow.glsl]: The global shadow library is for sampling the shadow map according to the user configuration (pcf, etc). It's not for atmosphere calculation which needs to be isolated in the atmosphere library. With this code sunlight would randomly reach underground during rain.

**Comment by supsm** [comment]
ceiling cloud is cool how dare you insult ceiling cloud :irritatered:

**Comment by supsm** [comment]
What

**Pull request merged**